icu4c/source/test/intltest/utfiteratortest.cpp

```diff
@@ -0,0 +1,1298 @@
+    0x7ff,
+    0x800,
+    0xfff,
+    0x1000,
+    0xd7ff,
+    Part(BAD16, 0xd800),
+    Part(BAD16, 0xdbff),
+    u'カ',
+    Part(BAD16, 0xdc00),
+    Part(BAD16, 0xdfff),
+    0xe000,
+    0xfffd,
+    0xffff,
+    0x10000,
+    0x10ffff,
+    Part(BAD32, 0x110000),
+    Part(BAD32, -1)
+};
+
+}  // namespace
+
+void UTFIteratorTest::initLong() {
+    if (!longGood32.str.empty()) { return; }
+    for (auto part : testParts) {
+        switch (part.type_) {
+        case GOOD: {
+            char u8[4];
+            size_t len8 = 0;
+            U8_APPEND_UNSAFE(u8, len8, part.c_);
+            longGood8.str.append(u8, len8);
+            longGood8.parts.push_back({u8, len8});
+            longBad8.str.append(u8, len8);
+            longBad8.parts.push_back({u8, len8});
+            longBad8.codePoints.push_back(part.c_);
+
+            char16_t u16[2];
+            size_t len16 = 0;
+            U16_APPEND_UNSAFE(u16, len16, part.c_);
+            longGood16.str.append(u16, len16);
+            longGood16.parts.push_back({u16, len16});
+            longBad16.str.append(u16, len16);
+            longBad16.parts.push_back({u16, len16});
+            longBad16.codePoints.push_back(part.c_);
+
+            longGood32.str.push_back(part.c_);
+            longGood32.parts.push_back({&part.c_, 1});
+            longBad32.str.push_back(part.c_);
+            longBad32.parts.push_back({&part.c_, 1});
+            longBad32.codePoints.push_back(part.c_);
+            break;
+        }
+        case BAD8: {
+            char u8[3] = {
+                static_cast<char>(part.u0_),
+                static_cast<char>(part.u1_),
+                static_cast<char>(part.u2_)
+            };
+            longBad8.str.append(u8, part.len_);
+            longBad8.parts.push_back({u8, part.len_});
+            longBad8.codePoints.push_back(U'?');
+            break;
+        }
+        case BAD16: {  // surrogate code unit / code point
+            char16_t u16 = part.u0_;
+            longBad16.str.push_back(u16);
+            longBad16.parts.push_back({&u16, 1});
+            longBad16.codePoints.push_back(U'?');
+            char32_t u32 = part.u0_;
+            longBad32.str.push_back(u32);
+            longBad32.parts.push_back({&u32, 1});
+            longBad32.codePoints.push_back(U'?');
+            break;
+        }
+        case BAD32: {
+            char32_t u32 = part.u0_;
+            longBad32.str.push_back(u32);
+            longBad32.parts.push_back({&u32, 1});
+            longBad32.codePoints.push_back(U'?');
+            break;
+        }
+        }
+    }
+    longGood8.codePoints = longGood16.codePoints = longGood32.codePoints = longGood32.str;
+}
+
+template<TestMode mode, UTFIllFormedBehavior behavior, IterType type, typename Unit, typename Units>
+void UTFIteratorTest::checkUnits(
+        const Units &units, std::basic_string_view<Unit> part, UChar32 expectedCP) {
+    // printf("U+%04lx\n", (long)units.codePoint());
```

Comment:
Are the commented-out printfs meant to be here (and in the following function)?

Reply:
removing. cleanup commit to follow.